packages/react-components/src/components/ImageGallery.tsx

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  DefaultButton,
+  IModalStyleProps,
+  IModalStyles,
+  IOverlayStyleProps,
+  IOverlayStyles,
+  IStyle,
+  IStyleFunctionOrObject,
+  Icon,
+  IconButton,
+  Layer,
+  Modal,
+  Stack,
+  mergeStyles
+} from '@fluentui/react';
+
+import React from 'react';
+import { BaseCustomStyles } from '../types';
+import {
+  cancelIcon,
+  closeButtonStyles,
+  controlBarContainerStyle,
+  downloadButtonStyle,
+  downloadIcon,
+  downloadIconStyle,
+  focusTrapZoneStyle,
+  headerStyle,
+  imageStyle,
+  overlayStyles,
+  scrollableContentStyle,
+  titleBarContainerStyle,
+  titleStyle
+} from './styles/ImageGallery.style';
+import { useTheme } from '../theming/FluentThemeProvider';
+import { isDarkThemed } from '../theming/themeUtils';
+
+/**
+ * Fluent styles for {@link ImageGallery}.
+ *
+ * @beta
+ */
+export interface ImageGalleryStylesProps extends BaseCustomStyles {
+  /** Styles for the ImageGallery Modal. */
+  modal?: IStyleFunctionOrObject<IModalStyleProps, IModalStyles>;
+  /** Styles for the ImageGallery Modal overlay. */
+  overlay?: IStyleFunctionOrObject<IOverlayStyleProps, IOverlayStyles>;
+  /** Styles for the ImageGallery header bar. */
+  header?: IStyle;
+  /** Styles for the ImageGallery titleBar container. */
+  titleBarContainer?: IStyle;
+  /** styles for the title label */
+  title?: IStyle;
+  /** Styles for the ImageGallery controlBar container. */
+  controlBarContainer?: IStyle;
+  /** Styles for the download button. */
+  downloadButton?: IStyle;
+  /** Styles for the close modal icon. */
+  closeIcon?: IStyle;
+  /** Styles for the image. */
+  image?: IStyle;
+}
+
+/**
+ * Props for {@link ImageGallery}.
+ *
+ * @beta
+ */
+export interface ImageGalleryImageProps {
+  imageUrl: string;
+  downloadFileName: string;
+  altText?: string;
+}
+
+/**
+ * Props for {@link ImageGallery}.
+ *
+ * @beta
+ */
+export interface ImageGalleryProps {
+  images: Array<ImageGalleryImageProps>;
+  onDismiss: () => void;
+  onImageDownloadButtonClicked: (imageUrl: string, downloadFileName: string) => void;
+  modalLayerHostId?: string;
+  title?: string;
+  titleIcon?: JSX.Element;
+  /**
+   * Allows users to pass in an object contains custom CSS styles.
+   * @Example
+   * ```
+   * <ImageGallery styles={{ image: { background: 'blue' } }} />
+   * ```
+   */
+  styles?: ImageGalleryStylesProps;
+}
+
+/**
+ * @beta
+ */
+export const ImageGallery = (props: ImageGalleryProps): JSX.Element => {
+  const { images, modalLayerHostId, onImageDownloadButtonClicked, onDismiss, title, titleIcon, styles } = props;
+  const theme = useTheme();
+  const isDarkTheme = isDarkThemed(theme);
+
+  const downloadButtonTitleString = 'Download';
+  const closeString = 'Close';
+  return (
+    <Layer hostId={modalLayerHostId}>
+      <Modal
+        titleAriaId={title}
+        isOpen={images.length > 0}
+        onDismiss={onDismiss}
+        isBlocking={false}
```

Comment:
sorry what's isBlocking and why it's set to false?

Reply:
> sorry what's isBlocking and why it's set to false?

It is for determining whether the dialog can be light dismissed by clicking outside the dialog (on the overlay).
I just removed it since it's defaulted to false